src/core/ComboBox/ComboBox.tsx

```diff
@@ -54,6 +55,20 @@ export type ComboBoxProps<T> = {
    * @default 'No options found'
    */
   emptyStateMessage?: string;
+  /**
+   * A custom item renderer can be specified to control the rendering.
+   * This function should ideally return a customized version of `MenuItem`,
+   * otherwise you will need to make sure to provide styling for the `isFocused` state.
+   */
+  itemRenderer?: (
+    option: SelectOption<T>,
+    states: {
```

Comment:
Move it to separate type because I think `itemRenderer` should be memoized and therefore users won't have IDE for the types as in your story:
![image](https://user-images.githubusercontent.com/36186912/154926863-33d52b7a-49dd-4dc9-a9b6-44ca19c35049.png)

Reply:
I also think it should be renamed because `id` and `index` is not really a state 😀

Reply:
This seems to work
![image](https://user-images.githubusercontent.com/9084735/155216936-cf10d82d-048e-4b2e-ab02-1bcc73a74b9a.png)

The name of the parameter doesn't matter because it will not be used directly. I was originally going to call it `options` but that got really confusing because of our menu options. Do you have a better suggestion?

Reply:
Doesn't look nice 😀 In `Select` I just named it `ItemRendererProps`

Reply:
so this one cannot be called `ItemRendererProps` if we want it in the barrel, to avoid conflicts.

Idk, all those extra props feel unnecessary to me when typescript has all these utilities

Reply:
Yeah and I hate all those utilities. It just makes simple things hard to understand and it is not that comfortable to use. Name it `ComboboxItemRendererProps`, idk.